<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>745</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPlainTextEdit" name="input">
        <property name="font">
         <font>
          <family>JetBrains Mono</family>
         </font>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPlainTextEdit" name="output">
        <property name="font">
         <font>
          <family>JetBrains Mono</family>
         </font>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="keyLabel">
          <property name="text">
           <string>Key</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="keyLineEdit">
          <property name="font">
           <font>
            <family>JetBrains Mono</family>
           </font>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="generateButton">
        <property name="text">
         <string>Generate key</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item alignment="Qt::AlignRight">
         <widget class="QRadioButton" name="encodeRadioButton">
          <property name="text">
           <string>Encode</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="convertButton">
          <property name="maximumSize">
           <size>
            <width>237</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Convert</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignLeft">
         <widget class="QRadioButton" name="decodeRadioButton">
          <property name="text">
           <string>Decode</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>convertButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>convertButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>generateButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>generateButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>512</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>encodeRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setEncodeMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>205</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>decodeRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>setDecodeMode()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>539</x>
     <y>546</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>input</sender>
   <signal>textChanged()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleConvertButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>372</x>
     <y>119</y>
    </hint>
    <hint type="destinationlabel">
     <x>372</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>convertButton()</slot>
  <slot>setEncodeMode()</slot>
  <slot>setDecodeMode()</slot>
  <slot>toggleConvertButton()</slot>
  <slot>generateButton()</slot>
 </slots>
</ui>
